Cheatsheet.md

```diff
@@ -0,0 +1,504 @@
+### [1. map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)
+
+**Description:**
+Creates an array by calling a specific function on each element present in the parent array.
+
+**Syntax:**
+* _array.map(function(currentValue, index, arr), thisValue)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| function(currentValue,index,arr) | Required function to be run for each array element |
+| thisValue | Optional. |
+
+**Return value:** 
+An Array containing the results of calling the provided function for each element in the original array.
+
+### [2. filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)
+
+**Description:**
+Creates an array filled with all array elements that pass a test (provided as a function).
+
+**Syntax:**
+ * _array.filter(function(currentValue, index, arr), thisValue)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| function(currentValue,index,arr) | Required function to be run for each array element |
+| thisValue | Optional. |
+
+**Return value:**
+Array containing all elements that pass the test else an empty array.
+
+**Example:**
+```
+let filtered = [112, 52, 0, -1, 944].filter(value => value>0);
+console.log(filtered);
+ ```
+*Output:*
+
+    [112,52,944]
+    
+### [3. reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)
+
+**Description:**
+reduces the array to a single value and executes provided function for each value of the array (from left-to-right).
+
+**Syntax:**
+ * _array.reduce( function(total, currentValue, currentIndex, arr), initialValue )_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| function(currentValue,index,arr) | Required function to be run for each array element |
+|initialValue | Optional.|
+
+**Return value:**
+returns accumulated result fron the last call of the callback function.
+
+**Example:**
+```
+const array1 = [1, 2, 3, 4];
+const reducer = (accumulator, currentValue) => accumulator + currentValue;
+console.log(array1.reduce(reducer));
+console.log(array1.reduce(reducer, 5));
+```
+*Output:*
+    
+    10
+    15
+
+### [4. slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)
+
+**Description:**
+Returns the selected elements in an array, as a new array object and selects the elements starting at the given start argument, and ends at the given end argument(excluding end argument).
+
+**Syntax:**
+ * _array.slice(start, end)_
+
+ | Parameter | Description |
+ | :-------- | :---------- |
+ | start     | Optional. specifies starting index of the selection    |
+ | end       | Optional. specifies ending index of the selection    |
+
+ **Return value:**
+ New array containing the selected elements.
+
+ **Example:**
+```
+    let arr = [23,56,87,32,75,13];
+    let new_arr = arr.slice();
+    console.log(arr);
+    console.log(new_arr);

```

Comment:
do not put console.log in examples. it's visual noise

Reply:
Sir, I am unable to understand. Do you mean to say that there is no need to print the output?

Reply:
Got it.